code/modules/food_and_drinks/drinks/drinks/shotglass.dm

```diff
@@ -20,9 +28,78 @@
 				filling.icon_state = "[icon_state]5"
 			if(80 to INFINITY)
 				filling.icon_state = "[icon_state]12"
-
 		filling.icon += mix_color_from_reagents(reagents.reagent_list)
 		overlays += filling
 		name = "shot glass of " + reagents.get_master_reagent_name() //No matter what, the glass will tell you the reagent's name. Might be too abusable in the future.
+		if(burn_state == ON_FIRE)
+			overlays += "shotglass_fire"
+			name = "flaming [name]"
 	else
 		name = "shot glass"
+
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/shotglass/proc/clumsilyDrink(mob/living/carbon/human/user) //Clowns beware
+	if(burn_state != ON_FIRE)
+		return
+	user.visible_message("<span class = 'warning'>[user] pours [src] all over themself!</span>", "<span class = 'danger'>You pour [src] all over yourself!</span>", "<span class = 'warning'>You hear a 'whoompf' and a sizzle.</span>")
+	reagents.reaction(user, TOUCH)
+	reagents.clear_reagents()
+	user.IgniteMob()
+	extinguish(silent = TRUE)
+
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/shotglass/proc/isShotFlammable()
+	var/datum/reagent/R = reagents.get_master_reagent()
+	if(istype(R, /datum/reagent/consumable/ethanol))
+		var/datum/reagent/consumable/ethanol/A = R
+		if(A.volume >= 5 && A.alcohol_perc >= 0.35) //Only an approximation to if something's flammable but it will do
+			return TRUE
+
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/shotglass/fire_act(global_overlay = FALSE, var/silent = FALSE)
+	if(!isShotFlammable() || burn_state == ON_FIRE) //You can't light a shot that's not flammable!
```

Comment:
Not sure why `fire_act` still runs through the whole proc proc even when the object is on fire. Overriden here